src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1672,37 +1675,125 @@ public override void VisitMarkupMinimizedTagHelperAttribute(MarkupMinimizedTagHe
                 {
                     foreach (var associatedDescriptor in associatedDescriptors)
                     {
-                        var associatedAttributeDescriptor = associatedDescriptor.BoundAttributes.First(a =>
+                        if (TagHelperMatchingConventions.TryGetFirstBoundAttributeMatch(
+                            attributeName,
+                            associatedDescriptor,
+                            out var associatedAttributeDescriptor,
+                            out var indexerMatch,
+                            out var parameterMatch,
+                            out var associatedAttributeParameterDescriptor))
                         {
-                            return TagHelperMatchingConventions.CanSatisfyBoundAttribute(attributeName, a);
-                        });
+                            var expectsBooleanValue = associatedAttributeDescriptor.ExpectsBooleanValue(attributeName);
 
-                        var expectsBooleanValue = associatedAttributeDescriptor.ExpectsBooleanValue(attributeName);
+                            if (!expectsBooleanValue)
+                            {
+                                // We do not allow minimized non-boolean bound attributes.
+                                return;
+                            }
```

Comment:
Why does this bail during lowering? I guess this is similar to the existing tag helpers logic.

Is the idea to not pass through something that's obviously wrong to the IR phase?

Reply:
Yup we simply short circuit that attribute and it never shows up in the IR.